c_src/exec.cpp

```diff
@@ -859,11 +863,12 @@ int process_command()
     return 0;
 }
 
-void initialize(int userid, bool use_alt_fds)
+void initialize(int userid, bool use_alt_fds, bool run_as_root)
 {
     // If we are root, switch to non-root user and set capabilities
-    // to be able to adjust niceness and run commands as other users.
-    if (getuid() == 0) {
+    // to be able to adjust niceness and run commands as other users,
+    // unless run_as_root is set
+    if (getuid() == 0 && !run_as_root) {
         superuser = true;
```

Comment:
Ah yes, that. I'm wondering if we shouldn't just ditch the `superuser` flag altogether in a separate PR before this gets merged?